meetings/2021-06-16.md

```diff
@@ -0,0 +1,151 @@
+# 2021-06-16 Solid Authorization
+
+https://meet.jit.si/solid-authorization
+
+
+## Agenda
+
+* Minutes https://github.com/solid/authorization-panel/pull/221
+    * Previous week actions
+* Follow up on previous week's UMA discussion
+* [UC3 Inheritance](https://github.com/solid/authorization-panel/)
+* [JWK in Access Control](https://github.com/solid/authorization-panel/discussions/225)
+* [Proofs and Origins](https://github.com/solid/authorization-panel/discussions/223)
+
+## Present
+
+* Aaron Coburn
+* Henry Story
+* Justin B
+* Elf P
+* Matthieu B
+* Eric P
+* Martynas Jusevicius
+
+
+## Minutes
+
+### UMA discussion
+
+Justin: [last week](2021-06-09.md) we went through slides for consent-based workflows. Given an agent/actor who want to give access to another app/user. A trusted authorization agent helps execute these workflows. This is a general pattern, but the interaction/protocol hasn't been specified.
+
+Justin: the interop spec just states: "apply permissions". A working, deployed private implementation with custom code (not using UMA, just secure redirects).
+
+Justin: UMA is an implementation of this pattern with good adoption. Designed to work with the OAuth2 stack and complements some of the other standards we are using in Solid.
+
+Justin: this is something we can look at as a trusted pattern. Not saying "UMA or bust", but thinks we should look at the problems it can help us with.
+
+Henry: What I found is that we have two things doing Authorization: the guard on the server, and this other Authorization server. This seems confusing. Why do we need two authorization systems?
+
+
+On the other hand thinking about the [launcher app](https://github.com/solid/authorization-panel/blob/main/proposals/LauncherApp.md) I proposed a couple of years ago, I could see how there are two Authorization roles:
+  * the Guard on the server, authorizing access to the resource
+  * the Guard near the wallet (essentially the Launcher App) authorizing when to sign headers, give out tokens or provide credentials
+  
+Henry: on the one side, you may have a wallet; on the other you verify the credential. Is that what UMA is trying to do? 
+
+Aaron: First of all if you have an authorization server like UMA (just a specialisation of OAuth 2), you can actually dispense the authorization guard on your AS. It leads to a much cleaner separation of concern. All you need is that the token would be valid for one specific request.
+
+Henry: I am very skeptical about this. There are reasons to be skeptical, as OAuth hardly works for our use case of decentralised hyper apps fetching data by following links. (In fact it seems to me more like  a catastrophe, a point that was echoed by some in last week's discussion on the Credentials CG). This is requiring me to have to read UMA and analyse why it might be wrong or if it is right somehow. That's a lot of work.
```

Comment:
```suggestion
Henry: I am very skeptical about this. There are reasons to be skeptical, as OAuth hardly works for our use case of decentralized hyper apps fetching data by following links. (In fact, it seems to me more like a catastrophe, a point that was echoed by some in last week's discussion on the Credentials CG). This is requiring me to have to read UMA and analyze why it might be wrong or if it is right somehow. That's a lot of work.
```